src/twisted/test/test_defer.py

```diff
@@ -680,6 +680,18 @@ def test_maybeDeferredSyncError(self):
         self.assertEqual(len(E), 1)
         self.assertEqual(str(E[0].value), expected)
 
+    def test_maybeDeferredSyncFailure(self):
+        """
+        L{defer.maybeDeferred} should catch exception raised by a synchronous
+        function and errback its resulting L{defer.Deferred} with it.
+        """
+        S, E = [], []
+        d = defer.maybeDeferred((lambda x: defer.fail(x + 5)), 10)
```

Comment:
I don't think that this matches the PR description. The `lambda` returns a `Deferred`, not a `Failure`.

Additionally, on Python 3 exceptions must derive from `BaseException`. `Failure(15)` creates a broken `Failure` object:

```
>>> Failure(15).raiseException()
Traceback (most recent call last):
  File "<console>", line 1, in <module>
  File "/home/twm/dev/twisted/src/twisted/python/failure.py", line 475, in raiseException
    raise self.value.with_traceback(self.tb)
AttributeError: 'int' object has no attribute 'with_traceback'
```

Perhaps this is a defect in `Failure`? It should probably coerce to `TypeError` internally, as you get if you try to raise a non-exception:

```
>>> raise 1
Traceback (most recent call last):
  File "<console>", line 1, in <module>
TypeError: exceptions must derive from BaseException
```

Reply:
Huh, right, ok. Rewriting.

Reply:
done